public/apps/configuration/panels/role-edit.tsx

```diff
@@ -14,26 +14,41 @@
  */
 
 import {
+  EuiAccordion,
```

Comment:
With your current change, this file is close to 500 lines with 3 major components, cluster permissions, index permissions and tenants. (the name one is relatively simple). The file has mixed constant variables, interfaces and util functions which are only specific to one component. This makes it difficult to mange. When I read it, I have to use cmd+f on the webapge to keep searching where those items are being used. I feel that this is a proper timing to extract major components out to their own files. If you want to do it in stages, probably start with the new code you added.

Reply:
Agreed, didn't think it could be this large at very beginning.